src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -634,84 +634,166 @@ private static object SplitOperatorImpl(ExecutionContext context, IScriptExtent
                 options |= SplitOptions.IgnoreCase;
             }
 
-            if (predicate != null)
+            if (predicate == null)
+            {
+                return SplitWithPattern(context, errorPosition, content, separatorPattern, limit, options);
+            }
+            else if (limit >= 0)
             {
                 return SplitWithPredicate(context, errorPosition, content, predicate, limit);
             }
             else
             {
-                return SplitWithPattern(context, errorPosition, content, separatorPattern, limit, options);
+                return NegativeSplitWithPredicate(context, errorPosition, content, predicate, limit);
             }
         }
 
-        private static object SplitWithPredicate(ExecutionContext context, IScriptExtent errorPosition, IEnumerable<string> content, ScriptBlock predicate, int limit)
+        private static object NegativeSplitWithPredicate(ExecutionContext context, IScriptExtent errorPosition, IEnumerable<string> content, ScriptBlock predicate, int limit)
```

Comment:
Any reason for `object` vs `IReadOnlyList<string>` or similar?

Reply:
No, I think that's just how it was when I got the code so I left it. Good call though.